### PR TITLE
Fix some DMARC issues

### PR DESCRIPTION
**Pull request by dasJ**
These issues were reported in #2872 which was closed by the stale bot.
This addresses the first two issues by getting rid of the string
"softfail" and by fixing the reason.

### CODE REVIEW

**Comment by vstakhov** [comment]
@fatalbanana looking good?

**Comment by fatalbanana** [comment]
The actual disposition is probably `none` in this case (not `quarantine`).

I have doubts about `<type>...</type>` inside `<reason>`- does the standard really call for it?

Lua doesn't have ternary operator so I suppose this needs to look slightly different.

I suppose it is most fitting to call `dmarc_report()` with `policy.dmarc_policy` rather than `disposition` which gets used in strange ways.

**Comment by flowerysong** [comment]
> I have doubts about `<type>...</type>` inside `<reason>`- does the standard really call for it?

Yes, it does.

```
   <xs:complexType name="PolicyOverrideReason">
     <xs:all>
       <xs:element name="type" type="PolicyOverrideType"/>
       <xs:element name="comment" type="xs:string"
                   minOccurs="0"/>
     </xs:all>
   </xs:complexType>
```

> I suppose it might be most fitting to call `dmarc_report()` with `policy.dmarc_policy` rather than `disposition` which gets used in strange ways (_unless_ we are indeed supposed to be reporting our effective disposition & not the one suggested by policy).

The report is supposed to include both the policy published by the domain (in `policy_published`) and the effective disposition after local policies are applied.

**Comment by dasJ** [comment]
@fatalbanana I rebased the branch to fix the tenary operator (also factored it into a variable to make things clearer).

**Comment by dasJ** [comment]
Also, since I don't know lua that well, does anyone have any hints about the third problem in #2872? The code needs to skip reports without `record` elements.

File: src/plugins/lua/dmarc.lua [hunk 1]
```diff
@@ -218,9 +218,10 @@ local function dmarc_report(task, spf_ok, dkim_ok, disposition,
   local dkim_fail = table.concat(dres.fail or E, '|')
   local dkim_temperror = table.concat(dres.temperror or E, '|')
   local dkim_permerror = table.concat(dres.permerror or E, '|')
+  local disposition_to_return = if disposition == "softfail" then "none" else disposition
```
Comment (vstakhov) [inline src/plugins/lua/dmarc.lua:221]: That won't work as well I'm afraid.

Comment (vstakhov) [reply src/plugins/lua/dmarc.lua:221]: `disposition == "softfail" and "none" or disposition` should work fine

**Comment by prehor** [comment]
Thanks for fixing #2872. 

Finally, for DMARC reports to be usable, emails with a score higher than Probably Spam must not be included in DMARC reports, as described in #2745, which was unfortunately closed by stale bot.

**Pull request merged**